modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonGramsTokenFilterFactory.java

```diff
@@ -58,5 +59,10 @@ public TokenStream create(TokenStream tokenStream) {
             return filter;
         }
     }
+
+    @Override
+    public TokenFilterFactory getSynonymFilter(boolean lenient) {
+        return IDENTITY_FILTER;
```

Comment:
So what happen if a rule contains a common term that produces an ngram ? It will never match ?

Reply:
For index-time common grams, the original tokens get emitted as well as the gram, so we can use IDENTITY_FILTER and ignore grams.  At query time, we need to use a CommonGramsQueryFilter which *only* emits the grams.

Reply:
Having tried to write a test for this, I've changed my mind.  We should throw errors here - common grams should never be placed before synonyms.

Reply:
+1